src/components/Layout/RightPaneLayout/RightPaneContext.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ReactElement, ReactNode, useContext, useMemo, useState } from "react";
+
+interface OpenRightPaneOpts {
+  content: ReactNode;
+}
+
+export type RightPaneLayoutContextProps = {
+  openInPane: (opts: OpenRightPaneOpts) => void;
+  closePane: () => void;
+  isRightPaneOpen: Boolean;
+  rightPaneContent: ReactNode;
+};
+
+export const RightPaneContext = React.createContext<RightPaneLayoutContextProps>({
+  openInPane: () => {},
+  closePane: () => {},
+  isRightPaneOpen: false,
+  rightPaneContent: null,
+});
+
+export function RightPaneProvider({ children }: { children: ReactElement }) {
+  const [rightPaneContent, setRightPaneContent] = useState<ReactNode>(undefined);
+  const [isRightPaneOpen, setIsRightPaneOpen] = useState<Boolean>(false);
+
+  const context = useMemo(() => {
+    return {
+      openInPane: (opts: OpenRightPaneOpts) => {
+        setRightPaneContent(opts?.content);
+        setIsRightPaneOpen(true);
+      },
+      closePane: () => setIsRightPaneOpen(false),
```

Comment:
**thought/suggestion:** What do you think about memoizing the `openInPane` and `closePane` methods separately? Right now they'll be built new every time a dependency changes, which _could_ have other implications further down the tree.
I'd think we could do something like:
```
const openInPane = useCallback(() => ..., [setRightPaneContent]);
const closePane = useCallback(() => ..., [setRightPaneContent]);

const contextValue = useMemo(() => ({ openInPane, closePane, rightPaneContent }), [openInPane, closePane, rightPaneContent]);
```
This way, the `openInPane` and `closePane` methods would only ever be rebuilt when if the setter method changes (which should be never).